[ios][docs] Update background modes configuration for iOS in multiple SDK pages (#32647)
expo/expo: docs/pages/versions/unversioned/sdk/background-fetch.mdx

`BackgroundFetch` only works when the app is backgrounded, not if the app was terminated or upon device reboot.
You can check out [the relevant GitHub issue](https://github.com/expo/expo/issues/3582) for more details.

On iOS the `BackgroundFetch` library requires you to use a [development build](/develop/development-builds/introduction/) since Background Fetch is not enabled in the iOS Expo Go app.

## Installation

<APIInstallSection />

## Configuration&ensp;<PlatformTags platforms={['ios']} />

To be able to run background fetch tasks on iOS, you need to add the `fetch` value to the `UIBackgroundModes` array in your app's **Info.plist** file. This is required for background fetch to work properly.

> **If you're using [CNG](/workflow/continuous-native-generation/)** the required `UIBackgroundModes` configuration will be applied automatically the next time you run `npx expo prebuild`.

Reviewer: ```suggestion
**If you're using [CNG](/workflow/continuous-native-generation/)**, the required `UIBackgroundModes` configuration will be applied automatically by prebuild.
```

slight preference for this wording because i want to avoid people misinterpreting that they need to run prebuild on their machine if they aren't doing that already (often folks will just use eas build)
